sqlmesh/core/model/definition.py

```diff
@@ -1468,4 +1468,7 @@ def _is_udtf(expr: exp.Expression) -> bool:
     "tags": lambda value: (
         exp.to_identifier(value[0]) if len(value) == 1 else exp.Tuple(expressions=value)
     ),
+    "grain": lambda value: (
+        exp.to_column(value[0]) if len(value) == 1 else exp.Tuple(expressions=exp.to_column(value))
+    ),
```

Comment:
Sorry for the late response, I just saw the discussion in the previous comment.

I'm not sure about the else clause here. If `value`'s supposed to be a list (which is consistent with the `len(value) == 1` check), then the `expressions=exp.to_column(value)` part doesn't really make sense in this context, because we can't convert a list to a column.

I think we might want to do something like:

```python
exp.Tuple(expressions=[exp.to_column(v) for v in value]
```

Same goes for the `"tags"` convert right above this.

cc: @tobymao

Reply:
Yep